features/product/managing_products/product_validation.feature

```diff
@@ -143,6 +143,15 @@ Feature: Products validation
         Then I should be notified that slug has to be unique
         And product with code "7-WONDERS-BABEL" should not be added
 
+    @ui
+    Scenario: Trying to remove price from an existing simple product while disabling channel
+        Given the store has a "Dice Brewing" product
+        And I want to modify this product
+        When I remove its price from channel "Web"
+        And I disable it in channel "Web"
+        And I try to save my changes
+        Then I should be notified that price cannot be blank
```

Comment:
I think I'm missing something here - if a product is disabled in a given channel, then the price for that channel shouldn't be required.

Reply:
true, but when a product has a price in channel, 
then you disable product in this channel and remove its price 
a notNullException is thrown. 
are you suggesting that in this case channelPricing record should be removed instead of being updated with null @pamil ?

Reply:
true, but when a product has a price in channel,
then you disable product in this channel and remove its price
a notNullException is thrown.
are you suggesting that in this case channelPricing record should be removed instead of being updated with null @pamil ?

Reply:
When is that NotNullException thrown?

Reply:
create a simple product and enable it for web channel and give it price in this channel 
after a product  is created, disable it from channel and remove price (set it empty)
then click save changes.
without this PR, a NotNullException should be thrown.

Reply:
create a simple product and enable it for web channel and give it price in this channel
after a product is created. update it by disable it from channel and remove price (set it empty)
then click save changes.
without this PR, a NotNullException should be thrown.

Reply:
I believe it's a description of the incorrect behavior we have in Sylius right now :( We should prevent this `NotNullException` for sure, but also give a possibility to remove the product price if it's possible